src/tests/system/tests/test_memory_cache.py

```diff
@@ -0,0 +1,1575 @@
+        2. Groups have correct names
+        3. SSSD is stopped
+        4. Groups are found
+        5. Groups have correct names
+    :customerscenario: False
+    """
+
+    def check(groups):
+        for group in groups:
+            result = client.tools.getent.group(group)
+            assert result is not None
+            assert result.name == group
+
+    groups = ["group1", "group2", "group3"]
+    for group in groups:
+        provider.group(group).add()
+
+    client.sssd.start()
+
+    check(groups)
+    client.sssd.stop()
+    check(groups)
+
+
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_memory_cache__disabled_passwd_getgrnam(client: Client, provider: GenericProvider):
+    """
+    :title: Lookup group by group name uses memory cache when SSSD is stopped and 'memcache_size_passwd' = 0
+    :setup:
+        1. Add 'group1', 'group2' and 'group3' to SSSD
+        2. In SSSD nss change 'memcache_size_passwd' to '0'
+        3. Start SSSD
+    :steps:
+        1. Find 'group1', 'group2' and 'group3' with getent.group(name)
+        2. Check that groups have correct names
+        3. Stop SSSD
+        4. Find 'group1', 'group2' and 'group3' with getent.group(name)
+        5. Check that groups have correct names
+    :expectedresults:
+        1. Groups are found
+        2. Groups have correct names
+        3. SSSD is stopped
+        4. Groups are found
+        5. Groups have correct names
+    :customerscenario: False
+    """
+
+    def check(groups):
+        for group in groups:
+            result = client.tools.getent.group(group)
+            assert result is not None
+            assert result.name == group
+
+    groups = ["group1", "group2", "group3"]
+    for group in groups:
+        provider.group(group).add()
+
+    client.sssd.nss["memcache_size_passwd"] = "0"
+    client.sssd.start()
+
+    check(groups)
+    client.sssd.stop()
+    check(groups)
+
+
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_memory_cache__disabled_passwd_getpwnam(client: Client, provider: GenericProvider):
+    """
+    :title: Lookup user by name when SSSD is stopped and 'memcache_size_passwd' = 0
+            uses memory cache therefore user is not found
+    :setup:
+        1. Add users to SSSD
+        2. Set users uids and gids
+        3. In SSSD nss change 'memcache_size_passwd' to '0'
+        4. Start SSSD
+    :steps:
+        1. Find 'user1', 'user2' and 'user3' with id(name)
+        2. Check that users have correct names
+        3. Stop SSSD
+        4. Find users with id(name)
+        5. Find users with id(uid)
+    :expectedresults:
+        1. Users are found
+        2. Users have correct names
+        3. SSSD is stopped
+        4. Users are not found
+        5. Users are not found
+    :customerscenario: False
+    """
+    ids = [("user1", 10001), ("user2", 10002), ("user3", 10003)]
+    for user, id in ids:
+        provider.user(user).add(uid=id, gid=id + 500)
```

Comment:
Technically, gid and uid can have the same value so there is no need to make them different. But it does not affect test.

Reply:
Yeah. I was wondering if it is possible. So thanks for the clarification. I will keep it this way, if it does not matter.